core/src/main/java/org/apache/spark/memory/TaskMemoryManager.java

```diff
@@ -101,29 +105,103 @@
   private final boolean inHeap;
 
   /**
+   * The size of memory granted to each consumer.
+   */
+  private final HashSet<MemoryConsumer> consumers;
```

Comment:
Add a comment to explain that this field is guarded by synchronizing on `this` (or use `@GuardedBy("this")`).